components/kyma-environment-broker/internal/director/oauth/client.go

```diff
@@ -58,12 +57,9 @@ func (c *oauthClient) WaitForCredentials() error {
 			Namespace: c.secretNamespace,
 			Name:      c.secretName,
 		}, secret)
-		switch {
-		case apierrors.IsNotFound(err):
+		if err != nil {
```

Comment:
it fails on `connection-refused` error on first call and it restarts our application.

Reply:
we still need to add initialDelayTimeout to our chart to completely fix this issue.